net/download/service/src/main/java/de/danoeh/antennapod/net/download/service/episode/DownloadAnnouncer.java

```diff
@@ -0,0 +1,36 @@
+package de.danoeh.antennapod.net.download.service.episode;
+
+import android.content.Context;
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityManager;
+import de.danoeh.antennapod.net.download.service.R;
+
+public class DownloadAnnouncer {
+    private final Context context;
+
+    public DownloadAnnouncer(Context context) {
+        this.context = context;
+    }
+
+    public void announceDownloadStart(String episodeTitle) {
+        String message = context.getString(R.string.download_started_talkback, episodeTitle);
+        announceDownloadStatus(message);
+    }
+
+    public void announceDownloadEnd(String episodeTitle, boolean success) {
```

Comment:
This method is never called with `success=false`

Reply:
I missed it :sweat_smile:

Adding it whenever sendErrorNotification happens.
There are few Result.failure() calls without notification. My understanding is those are not communicated to the user, although I'm not confident I'm not missing some handling of the failure happening somewhere else entirely.

Reply:
Now that you mention it, I think we should leave this out when posting an error notification anyway. Otherwise it gets announced twice (notification+TalkBack) and we don't want people to turn off error notifications. Opinions @keunes?

Reply:
> I think we should leave this out when posting an error notification anyway. Otherwise it gets announced twice (notification+TalkBack) and we don't want people to turn off error notifications.

Yeap, makes sense!

Reply:
If there's conflict with error notifications, I'd go with removing it for now and considering the interaction only partially covered.

My reasoning is there's more uncovered interactions leading to open ending for user anyway. For example 'cancel' gives no feedback - I didn't dig into it since it's out of scope for this specific issue, it's just something I noticed. I have the least decision making context here, but leaving 'ended(in failure)' in doesn't eliminate no-feedback end state for the user, so unambiguity is not achieved anyway.

Reply:
> what if users have notifications disabled (do not disturb)? Then it'll be just an open ending without info for the user. Can't think of a way around it. Do we just accept that risk?

I would say we accept that risk. We have the same for seeing users already, when they hide the notifications and an error occurs, they do not get notified.